substrate/frame/tx-pause/src/lib.rs

```diff
@@ -15,6 +15,48 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+//! # Transaction Pause
+//!
+//! Allows dynamic, chain-state-based pausing and unpausing of specific extrinsics via call filters.
+//!
+//! ## Pallet API
+//!
+//! See the [`pallet`] module for more information about the interfaces this pallet exposes,
+//! including its configuration trait, dispatchables, storage items, events and errors.
+//!
+//! ## Overview
+//!
+//! A dynamic call filter that can be controlled with extrinsics.
+//!
+//! Pausing an extrinsic means that the extrinsic CANNOT be called again until it is unpaused.
+//! The exception is calls that use `dispatch_bypass_filter`, typically only with the root origin.
+//!
+//! ### Primary Features
+//!
+//! - Calls that should never be paused can be added to a whitelist.
+//! - Separate origin configuration for pausing and pausing.
+//! - Pausing is triggered using the string representation of the call.
+//! - Pauses can target a single extrinsic or an entire pallet.
+//!
+//! ### Example
+//!
+//! Configuration of call filters:
```

Comment:
Please also another simple example of how to use it, possibly by `docify` embedding a test 🙏

Reply:
Tried setting this up. Does this look useful/correct to you?
![image](https://github.com/paritytech/polkadot-sdk/assets/1252199/e74f240c-0310-4467-b6ec-fb318d7501c9)

Reply:
Yea this looks good. Maybe some more comments in the tests, but i think it works for a first version 👍